src/main/java/duke/Main.java

```diff
@@ -0,0 +1,329 @@
+        //if you have already created the file
+        File dataFile = new File(String.valueOf(filePath));
+        try {
+            Scanner s = new Scanner(dataFile);
+            while (s.hasNext()) {
+                String[] words = s.nextLine().split("//");
+                switch (words[0]) {
+                    case "T":
+                        tasks.add(new Todo(words[2]));
+                        if (Integer.parseInt(words[1]) == 1) {
+                            tasks.get(tasks.size() - 1).markAsDone(true);
+                        }
+                        break;
+                    case "E":
+                        tasks.add(new Event(words[2], words[3]));
+                        if (Integer.parseInt(words[1]) == 1) {
+                            tasks.get(tasks.size() - 1).markAsDone(true);
+                        }
+                        break;
+                    case "D":
+                        tasks.add(new Deadline(words[2], words[3]));
+                        if (Integer.parseInt(words[1]) == 1) {
+                            tasks.get(tasks.size() - 1).markAsDone(true);
+                        }
+                        break;
+                    default:
+                        System.out.println("You edit the file in a wrong format. Please check.");
+                        break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("file not found");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("You edit the file in a wrong format. Please check.");
+        }
+        return filePath;
+    }
+
+    private static void writeFile(Path filePath) {
+        try {
+            FileWriter fw = new FileWriter(String.valueOf(filePath));
+            for (Task task : tasks) {
+                fw.write(task.fileString());
+                fw.write(System.lineSeparator());
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("something went wrong when writing to file");
+        }
+    }
+
+
+    private static Path createFolderAndFIle() {
+        String home = System.getProperty("user.dir");
+
+        Path folderPath = java.nio.file.Paths.get(home, "data");
+        Path filePath = java.nio.file.Paths.get(home, "data", "duke.txt");
+
+        if (!java.nio.file.Files.exists(folderPath)) {
+
+            try {
+                java.nio.file.Files.createDirectory(folderPath);
+            } catch (IOException e) {
+                System.out.println("The folder \"data\" is not created");
+            }
+        }
+        if (!java.nio.file.Files.exists(filePath)) {
+            try {
+                java.nio.file.Files.createFile(filePath);
+            } catch (IOException e) {
+                System.out.println("The file \"duke.txt\" is not created");
+            }
+        }
+        return filePath;
+    }
```

Comment:
Perhaps you can create a Storage class to store the methods and variables needed to save your files? This can improve the readability of your code